### PR TITLE
Use NodePattern to make Lint/Debugger more concise

### PR DESCRIPTION
**Pull request by alexdowad**
Benchmarking shows that the NodePattern matcher is faster than the previous
implementation.

### CODE REVIEW

File: lib/rubocop/cop/lint/debugger.rb [hunk 1]
```diff
@@ -7,71 +7,17 @@ module Lint
...
-          PRY_NODE,
-          REMOTE_PRY_NODE,
-          PRY_REMOTE_NODE,
-          PRY_RESCUE,
-          CAPYBARA_SAVE_PAGE,
-          CAPYBARA_SAVE_OPEN_SCREENSHOT,
-          CAPYBARA_SAVE_SCREENSHOT
-        ]
+        def_node_matcher :debugger_call?,
+                         '{(send nil {:debugger :byebug} ...)
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/debugger.rb]: Now that we've used node patterns a bit I know that I like them a lot, but I hate dealing with multiline strings. Ideally we'd be able to represent them is some more structured manner.

Comment (alexdowad) [inline lib/rubocop/cop/lint/debugger.rb]: Aha. Like an Array?

Comment (alexdowad) [inline lib/rubocop/cop/lint/debugger.rb]: I've discovered another thing. There is a need for syntax to represent the parent of the matched node...

Comment (alexdowad) [inline lib/rubocop/cop/lint/debugger.rb]: Hmm. Just thinking whether that is really a good idea or not.

**Pull request merged**